daemon/daemon_main.go

```diff
@@ -1454,3 +1413,111 @@ func (d *Daemon) instantiateAPI() *restapi.CiliumAPI {
 
 	return api
 }
+
+func initKubeProxyFreeOptions() {
+	if option.Config.KubeProxyReplacement != option.KubeProxyReplacementStrictPartial &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementPartial &&
+		option.Config.KubeProxyReplacement != option.KubeProxyReplacementProbe {
+		log.Fatalf("Invalid value for --%s: %s", option.KubeProxyReplacement, option.Config.KubeProxyReplacement)
+	}
+
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementProbe {
+		option.Config.EnableNodePort = true
+		option.Config.EnableExternalIPs = true
+		option.Config.EnableHostReachableServices = true
+		option.Config.EnableHostServicesTCP = true
+		option.Config.EnableHostServicesUDP = true
+	}
+
+	if option.Config.EnableNodePort && option.Config.EnableIPSec {
+		log.Fatal("IPSec cannot be used with NodePort BPF")
```

Comment:
Still not sure about this one. Maybe in a non-strict mode we should just log a warning, disable the NP feature, and proceed further?